powerstore/datasource_filesystem_snapshot.go

```diff
@@ -0,0 +1,151 @@
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ datasource.DataSource              = &fileSystemSnapshotDataSource{}
+	_ datasource.DataSourceWithConfigure = &fileSystemSnapshotDataSource{}
+)
+
+// newfileSystemSnapshotDataSource returns the fileSystemSnapshot snapshot data source object
+func newFileSystemSnapshotDataSource() datasource.DataSource {
+	return &fileSystemSnapshotDataSource{}
+}
+
+// fileSystemSnapshotDataSource is the data source implementation
+type fileSystemSnapshotDataSource struct {
+	client *client.Client
+}
+
+// Metadata returns the data source type name
+func (d *fileSystemSnapshotDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_filesystem_snapshot"
+}
+
+// Schema defines the schema for the data source
+func (d *fileSystemSnapshotDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description:         "This datasource is used to query the existing File System Snapshot from PowerStore array. The information fetched from this datasource can be used for getting the details for further processing in resource block.",
+		MarkdownDescription: "This datasource is used to query the existing File System Snapshot from PowerStore array. The information fetched from this datasource can be used for getting the details for further processing in resource block.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Unique identifier of the File System Snapshot. Conflicts with `name` and `filesystem_id`.",
+				MarkdownDescription: "Unique identifier of the File System Snapshot. Conflicts with `name` and `filesystem_id`.",
+				Optional:            true,
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("name"), path.MatchRoot("filesystem_id")),
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description:         "File System Snapshot name. Conflicts with `id` and `filesystem_id`.",
+				MarkdownDescription: "File System Snapshot name. Conflicts with `id` and `filesystem_id`.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.ConflictsWith(path.MatchRoot("id"), path.MatchRoot("filesystem_id")),
+				},
+			},
+
+			"filesystem_id": schema.StringAttribute{
+				Description:         "File System Snapshot name. Conflicts with `id` and `name`.",
+				MarkdownDescription: "File System Snapshot name. Conflicts with `id` and  `name`.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
```

Comment:
conflicts with validation is not present here. Please add it

Reply:
I have added the conflict validation. please verify